sqlserver/assets/configuration/spec.yaml

```diff
@@ -39,10 +39,56 @@ files:
       value:
         type: string
     - name: database
-      description: Database name to query
+      description: |
+        Database name to query.  Not compatible with `database_autodiscovery`.
       value:
         type: string
         example: master
+    - name: database_autodiscovery
+      description: |
+        Auto-discover and monitor databases.  If `true`, overrides `database` option.
+        Can be combined with `autodiscovery_include` and `autodiscovery_exclude` options.
+      value:
+        type: boolean
+        example: false
+    - name: autodiscovery_include
+      description: |
+        Regular expression for specific database names to always include as part of `database_autodiscovery`.
+        Will only report metrics for databases that are actually found in this instance, so missing
+        databases from this option should be harmless.
+
+        Character casing is ignored. For convenience, the regular expressions start matching from the beginning and
+        therefore to match anywhere you must prepend `.*`. For exact matches append `$`. Casing is ignored.
+
+        Defaults to `.*` to include everything.
+      value:
+        type: array
+        items:
+          type: string
+        example:
+          - master$
+          - AdventureWorks.*
+    - name: autodiscovery_exclude
+      description: |
+        Regular expression for specific database names to always exclude as part of `database_autodiscovery`.
+
+        Character casing is ignored. For convenience, the regular expressions start matching from the beginning and
+        therefore to match anywhere you must prepend `.*`. For exact matches append `$`. Casing is ignored.
+
+        When conflicts arise, excluded databases will take precedence over any found from `autodiscovery_include`.
```

Comment:
```suggestion
        In case of conflicts, database exclusion via `autodiscovery_exclude` takes precedence over those found via `autodiscovery_include`.
```